platform/include/tfm_hal_its.h

```diff
@@ -73,6 +78,120 @@ extern ARM_DRIVER_FLASH TFM_HAL_ITS_FLASH_DRIVER;
 enum tfm_hal_status_t
 tfm_hal_its_fs_info(struct tfm_hal_its_fs_info_t *fs_info);
 
+#ifdef TFM_ITS_ENCRYPTED
+
+/**
+ * \brief Generate an encryption nonce
+ *
+ * \details The nonce has to be unique for every encrypted file, even across
+ *          resets. Therefore it is recommended to base the nonce on a
+*           pseudorandom number.
```

Comment:
We use the word therefore. But I don't see how the second sentence follows from the first.

It's not clear to me what "base the nonce on" means in the context of this sentence either.

Reply:
Ok, any better suggestions? Then we just use that one ;)

Reply:
"The nonce must be unique for every encrypted file, even across resets."

AFAICT it is implementation defined whether or not it is a good idea to use a pseudorandom number to implement this, or a non-volatile monotonic counter.

This is based on the nonce not being a secret, which I don't know if it is or not.

Reply:
Ok, I thought it is nice to give people a hint, of what would be a good approach instead of leaving the fully on their own, but no strong will on this removed the second sentence.